### PR TITLE
reveiw1code

### PR DESCRIPTION
**Pull request by ToyeshShendeSyngenta**

### CODE REVIEW

File: src/components/FrontPage.js [hunk 1]
```diff
@@ -1,34 +1,162 @@
...
+      .get(apiUrl)
+      .then((response) => setLaunchData(response.data))
+      .catch((error) => console.error("Error fetching data: ", error));
+  }, [filters]);
+
+  const getFilterParams = () => {
+    const { launchSuccess, landSuccess, launchYear } = filters;
+    let params = "";
+
+    if (launchSuccess !== null) {
```
Comment (anjaliD-syn) [inline src/components/FrontPage.js:29]: can we avoid this check against null by !launchSuccess or something?

File: src/components/FrontPage.js [hunk 2]
```diff
@@ -1,34 +1,162 @@
...
-              <p><b>Mission IDs: </b> {launch.mission_id.join(', ')}</p>
-              <p><b>Launch Year: </b> {launch.launch_year}</p>
-              <p><b>Launch Success: </b> {launch.launch_success ? 'Yes' : 'No'}</p>
-              <p><b>Successful Landing : </b>{launch.rocket.first_stage.cores[0].land_success ? 'Yes' : 'No'}</p>
+        <div className="filters">
+          <h4>Filters</h4>
+          <div className="filter-section">
+            <div className="filter-subsection">
+              <h4>Name Search</h4>
+              <input
```
Comment (anjaliD-syn) [inline src/components/FrontPage.js]: can you please use antd input component here?

File: src/components/FrontPage.js [hunk 3]
```diff
@@ -1,34 +1,162 @@
...
-          ))}
+            <div className="filter-subsection1">
+              <h4>Successful Launch</h4>
+              <button
+                className={filters.launchSuccess === true ? "active" : ""}
+                onClick={() => handleFilterChange("launchSuccess", true)}
+              >
+                True
+              </button>
+              <button
```
Comment (anjaliD-syn) [inline src/components/FrontPage.js]: can you please use antd button component here?

File: src/components/FrontPage.js [hunk 4]
```diff
@@ -1,34 +1,162 @@
...
+                False
+              </button>
+            </div>
+          </div>
+        </div>
+        <div className="cards">
+          {launchData
+            .filter(
+              (launch) =>
+                search === "" ||
```
Comment (anjaliD-syn) [inline src/components/FrontPage.js]: Avoid checking against '',undefined or null values.

File: src/components/FrontPage.js [hunk 5]
```diff
@@ -1,34 +1,162 @@
...
+            <div className="filter-subsection1">
+              <h4>Successful Landing</h4>
+              <button
+                className={filters.landSuccess === true ? "active" : ""}
+                onClick={() => handleFilterChange("landSuccess", true)}
+              >
+                True
+              </button>
+              <button
+                className={filters.landSuccess === false ? "active" : ""}
```
Comment (anjaliD-syn) [inline src/components/FrontPage.js:122]: you can avoid adding empty('') classname here and also avoid boolean comparion, instead you can write 
filters.launchSuccess && 'active'

File: src/components/FrontPage.js [hunk 6]
```diff
@@ -1,34 +1,162 @@
...
+                className={filters.launchSuccess === false ? "active" : ""}
+                onClick={() => handleFilterChange("launchSuccess", false)}
+              >
+                False
+              </button>
+            </div>
+            <div className="filter-subsection1">
+              <h4>Successful Landing</h4>
+              <button
+                className={filters.landSuccess === true ? "active" : ""}
```
Comment (anjaliD-syn) [inline src/components/FrontPage.js]: you can avoid adding empty('') classname here and also avoid boolean comparion, instead you can write 
filters.landSuccess && 'active'

check at other places also.

**Review by anjaliD-syn — CHANGES_REQUESTED** [review]
resolve comments

File: src/components/FrontPage.js (hunk 1 above)
Comment (ToyeshShendeSyngenta) [reply src/components/FrontPage.js:29]: by using !lanchSuceess only true tag is working in the filter  becaue launchsucess is predefined as null and  its entering the if statement even the value is null and chaing the param  by     if (launchSuccess !== null) both filters are working as expected

**Review by anjaliD-syn — APPROVED** [review]

**Pull request merged**